src/server/pfs/server/driver.go

```diff
@@ -3036,21 +3070,22 @@ func (d *driver) deleteFile(pachClient *client.APIClient, file *pfs.File) error
 		if branch == "" {
 			return pfsserver.ErrCommitFinished{file.Commit}
 		}
-		_, err := d.makeCommit(pachClient, "", client.NewCommit(file.Commit.Repo.Name, ""), branch, nil, nil, []string{file.Path}, []*pfs.PutFileRecords{&pfs.PutFileRecords{Tombstone: true}}, "")
+		_, err := d.makeCommit(pachClient, stm, "", client.NewCommit(file.Commit.Repo.Name, ""), branch, nil, nil, []string{file.Path}, []*pfs.PutFileRecords{&pfs.PutFileRecords{Tombstone: true}}, "")
 		return err
 	}
+	// TODO: what do with this in STM transaction?
```

Comment:
Also this.

Reply:
I'm thinking I can just drop this TODO - `upsertPutFileRecords` doesn't appear to ever be done in a transaction, but I'm curious if this could cause any problems - @jdoliner?